### PR TITLE
Junit5 migration

### PR DESCRIPTION
**Pull request by Frawless**
This PR introduce junit  migration from version 4 to 5. I changed some values for checkstyle. However, these values should be changed back after next test refactor.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -112,6 +139,38 @@
 
 	<build>
 		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
```
Comment (serrss) [inline pom.xml]: Do we need 2 same plugins in one POM.xml? `<artifactId>maven-surefire-plugin</artifactId>` line 144 and line 226

Comment (Frawless) [reply pom.xml]: Nope, going to remove it, thanks

File: .checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -87,21 +87,23 @@
         </module>
 
         <!-- code quality -->
-        <module name="MethodLength"/>
+        <module name="MethodLength">
```
Comment (scholzj) [inline .checkstyle/checkstyle.xml]: Can't we just suppress these with annotations directly in the code? That is much better then rewriting the rules for all methods.

Comment (Frawless) [reply .checkstyle/checkstyle.xml]: Make sense, let me change it

File: .checkstyle/checkstyle.xml [hunk 3]
```diff
@@ -87,21 +87,23 @@
...
+            <property name="max" value="15"/>
         </module>
         <module name="BooleanExpressionComplexity">
             <!-- default is 3 -->
         </module>
 
         <module name="ClassFanOutComplexity">
             <!-- default is 20 -->
-            <property name="max" value="36"/>
+            <property name="max" value="40"/>
```
Comment (ppatierno) [inline .checkstyle/checkstyle.xml]: I was always trying to not changing the default metrics value to suppress locally or fixing the problems why now we are doubling these values? :-)

File: src/test/java/io/strimzi/kafka/bridge/http/HttpBridgeTest.java [hunk 4]
```diff
@@ -2249,53 +2196,57 @@ public void seekToBeginningAndReceive(TestContext context) {
...
         kafkaCluster.createTopic(topic, 1, 1);
 
-        Async creationAsync = context.async();
+        CompletableFuture<Boolean> produce = new CompletableFuture<>();
+        CompletableFuture<Boolean> create = new CompletableFuture<>();
+        CompletableFuture<Boolean> subscribe = new CompletableFuture<>();
+        CompletableFuture<Boolean> delete = new CompletableFuture<>();
+        CompletableFuture<Boolean> seek = new CompletableFuture<>();
+        CompletableFuture<Boolean> dummy = new CompletableFuture<>();
+        CompletableFuture<Boolean> consumeSeek = new CompletableFuture<>();
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/HttpBridgeTest.java]: can we create them every time each one is needed?

Comment (Frawless) [reply src/test/java/io/strimzi/kafka/bridge/http/HttpBridgeTest.java]: Sure, I was thinking about that as well.

**Comment by serrss** [comment]
@scholzj Sorry, I'm not sure that I understand about what things you are talking about.

**Comment by serrss** [comment]
@scholzj oh, you mean 2 plugins POM file

**Comment by scholzj** [comment]
Exactly.

**Comment by ppatierno** [comment]
@Frawless I don't see any further work on this after my comments. Any news?

**Comment by Frawless** [comment]
@ppatierno  finishing this one https://github.com/strimzi/strimzi-kafka-operator/pull/1613, after that I will get back to this PR.

File: pom.xml [hunk 5]
```diff
@@ -112,6 +139,32 @@
 
 	<build>
 		<plugins>
+			<plugin>
```
Comment (scholzj) [inline pom.xml]: @Frawless I do not think you should be adding this. The assembly is done on L198-215.

Comment (Frawless) [reply pom.xml]: @scholzj redundant assembly removed

**Pull request merged**